console/views/logos.py

```diff
@@ -259,3 +262,33 @@ def post(self, request, *args, **kwargs):
                             kind="team", kind_id=team.tenant_id, user=user, role_ids=[developer.ID])
         result = general_message(msg="success", msg_show=u"初始化权限分配成功", code=200)
         return Response(result, status=200)
+
+
+class Upgrade(JWTAuthApiView):
+    def post(self, request, *args, **kwargs):
+        version = get_current_version()
+        sync_app = request.data.get("sync_app", False)
+        # When upgrading to version 5.3.0, applications need to be synchronized
+        if should_sync_app(version) or sync_app:
+            teams = team_repo.get_all_have_region_tenant()
+            err_regions = []
+            for team in teams:
+                if team.region in err_regions:
+                    continue
+                apps = group_repo.get_tenant_region_groups(team.tenant_id, team.region)
+                for app in apps:
+                    err_region = group_service.sync_app_services(team, team.region, app.ID)
```

Comment:
Upgrade data should be upgraded by region. You should not upgrade every application in a loop.